### PR TITLE
Clientslist

### PR DESCRIPTION
**Pull request by appgurueu**
for reviewing some WIP work

### CODE REVIEW

File: builtin/mainmenu/dlg_clients_list.lua [hunk 1]
```diff
@@ -0,0 +1,41 @@
+local function clients_list_formspec(dialogdata)
+	local clients_list = server.clients_list
```
Comment (appgurueu) [inline builtin/mainmenu/dlg_clients_list.lua]: Take it via the parameter:
```suggestion
	local clients_list = dialogdata.server.clients_list
```

File: builtin/mainmenu/dlg_clients_list.lua [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	local servername  = server.name
```
Comment (appgurueu) [inline builtin/mainmenu/dlg_clients_list.lua]:
```suggestion
	local servername  = dialogdata.server.name
```

File: builtin/mainmenu/dlg_clients_list.lua [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		local escaped = {}
+		for i, str in ipairs(clients_list) do
+			escaped[i] = core.formspec_escape(str)
+		end
+		return table.concat(escaped, ",")
+	end
+
+	local list = "textlist[1,1;7.75,9;client_list;" .. fmt_formspec_list(clients_list) .. "]"
+
+	local retval =
```
Comment (appgurueu) [inline builtin/mainmenu/dlg_clients_list.lua]:
```suggestion
	return
```

File: builtin/mainmenu/dlg_clients_list.lua [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+		"size[10,11]" ..
+		"button[3.5,10.25;3,0.8;dlg_clients_list_exit;OK]" ..
+		"label[4,0.5;This is the clientlist from " .. fgettext(servername) .. "]" ..
+		list
+	return retval
```
Comment (appgurueu) [inline builtin/mainmenu/dlg_clients_list.lua]:
```suggestion
```

File: builtin/mainmenu/dlg_clients_list.lua [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+end
+
+--------------------------------------------------------------------------------
+
+local function clients_list_buttonhandler(this, fields)
+	if fields["dlg_clients_list_exit"] then
```
Comment (appgurueu) [inline builtin/mainmenu/dlg_clients_list.lua]:
```suggestion
	if fields.dlg_clients_list_exit then
```

File: builtin/mainmenu/dlg_clients_list.lua [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+		this:delete()
+		return true
+	end
+	return false
+end
+
+--------------------------------------------------------------------------------
+
+function create_clientslist_dialog(server)
+	local retval = dialog_create("dlg_clients_list",
```
Comment (appgurueu) [inline builtin/mainmenu/dlg_clients_list.lua:53]:
```suggestion
	return dialog_create("dlg_clients_list",
```

File: builtin/mainmenu/tab_online.lua [hunk 7]
```diff
@@ -111,12 +111,40 @@ local function get_formspec(tabview, name, tabdata)
 		if gamedata.fav then
 			retval = retval .. "tooltip[btn_delete_favorite;" .. fgettext("Remove favorite") .. "]"
 			retval = retval .. "style[btn_delete_favorite;padding=6]"
-			retval = retval .. "image_button[5,1.3;0.5,0.5;" .. core.formspec_escape(defaulttexturedir ..
+			retval = retval .. "image_button[4.5,1.3;0.5,0.5;" .. core.formspec_escape(defaulttexturedir ..
 				"server_favorite_delete.png") .. ";btn_delete_favorite;]"
 		end
-		if gamedata.serverdescription then
+		if _G.gamedata.serverdescription then
```
Comment (appgurueu) [inline builtin/mainmenu/tab_online.lua]:
```suggestion
		if gamedata.serverdescription then
```

File: builtin/mainmenu/tab_online.lua [hunk 8]
```diff
@@ -111,12 +111,40 @@ local function get_formspec(tabview, name, tabdata)
...
 			retval = retval .. "textarea[0.25,1.85;5.25,2.7;;;" ..
-				core.formspec_escape(gamedata.serverdescription) .. "]"
+				core.formspec_escape(_G.gamedata.serverdescription) .. "]"
```
Comment (appgurueu) [inline builtin/mainmenu/tab_online.lua]:
```suggestion
				core.formspec_escape(gamedata.serverdescription) .. "]"
```

File: builtin/mainmenu/dlg_clients_list.lua [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+		clients_list_formspec,
+		clients_list_buttonhandler,
+		nil)
+	return retval
```
Comment (appgurueu) [inline builtin/mainmenu/dlg_clients_list.lua:58]: There should be something like a `retval.data.server = server` before this so you can use `dialogdata.server` above.

**Review by appgurueu — COMMENTED** [review]
first look

File: builtin/mainmenu/dlg_clients_list.lua [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+		this:delete()
+		return true
+	end
+	return false
+end
+
+--------------------------------------------------------------------------------
+
+function create_clientslist_dialog(server)
+	return dialog_create("dlg_clients_list",
```
Comment (appgurueu) [inline builtin/mainmenu/dlg_clients_list.lua]:
```suggestion
	local retval = dialog_create("dlg_clients_list",
```
sorry if the previous suggestion was wrong

**Review by siliconsniffer — APPROVED** [review]
Implemented =)

File: builtin/mainmenu/tab_online.lua
Comment (siliconsniffer) [inline builtin/mainmenu/tab_online.lua:1]: My editor inserted some whitespaces, the checks for that didn't throw any errors though. Is it fine like this?

Comment (appgurueu) [reply builtin/mainmenu/tab_online.lua:1]: It's not a big deal but ideally such small cleanups should not be mixed with features, they clutter the diff.

Comment (siliconsniffer) [reply builtin/mainmenu/tab_online.lua:1]: Ah, okay - will remember it for next time. Rest looks good? If so I'd create the PR now.

**Pull request closed without merging**